Tasks/AzureRmWebAppDeployment/FindInstalledMSDeploy.ps1

```diff
@@ -0,0 +1,146 @@
+﻿
+function Get-MSDeployOnTargetMachine
+{
+    try
+    {
+        $MSDeployPath, $MSDeployVersion = Locate-HighestVersionMsDeployPackage
+        $MSDeployVersionNumber = [decimal] $MSDeployVersion
+    }
+    catch [System.Exception]
+    {
+        Write-Verbose ("Failed to get MS Deploy location with exception: " + $_.Exception.Message)
+        $MSDeployVersionNumber = 0
+    }
+
+    if ($MSDeployPath -eq $null)
+    {
+        throw  "Unable to find the location of MS Deploy from registry on machine $env:COMPUTERNAME"
+    }
+    else
+    {
+        return $MSDeployPath, $MSDeployVersionNumber
+    }
+}
+
+function Get-RegistryValueIgnoreError
+{
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [Microsoft.Win32.RegistryHive]
+        $RegistryHive,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $Key,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $Value,
+
+        [parameter(Mandatory = $true)]
+        [Microsoft.Win32.RegistryView]
+        $RegistryView
+    )
+
+    try
+    {
+        $baseKey = [Microsoft.Win32.RegistryKey]::OpenBaseKey($RegistryHive, $RegistryView)
+        $subKey =  $baseKey.OpenSubKey($Key)
+        if($subKey -ne $null)
+        {
+            return $subKey.GetValue($Value)
+        }
+    }
+    catch
+    {
+    }
+    return $null
+}
+
+function Get-SubKeysInFloatFormat($keys)
+{
+    $targetKeys = @() 
+        foreach ($key in $keys) 
+        {		  
+            try {
+                $targetKeys += [decimal] $key
+            }
+            catch {}
+        }
+   
+    $targetKeys    
+}
+
+function Get-MSDeployPackage([int] $majorVersion, [bool] $wow6432Node)
+{
+    $MSDeployInstallRootRegKey = "SOFTWARE", "Microsoft", "IIS Extensions", "MSDeploy", "$majorVersion" -join [System.IO.Path]::DirectorySeparatorChar
```

Comment:
make sure verbose logs are enough ti check whether we are looking things for 32-bit or 64-bit.. applicable everywher